lib/components/SelectionControl/Checkbox/index.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+    ComponentProps,
+    forwardRef,
+    ReactNode,
+    useState,
+    useEffect,
+} from "react";
+import clsx from "clsx";
+import "./checkbox.scss";
+import {
+    StandaloneSquareCheckFillIcon,
+    StandaloneSquareBoldIcon,
+    StandaloneSquareMinusFillIcon,
+    LabelPairedCircleInfoSmRegularIcon,
+    LabelPairedCircleInfoMdRegularIcon,
+} from "@deriv/quill-icons";
+import { TMediumSizes } from "../../../types";
+import { Text } from "../../Typography";
+
+interface CheckboxProps
+    extends Omit<
+        ComponentProps<"input">,
+        "placeholder" | "defaultChecked" | "size"
+    > {
+    indeterminate?: boolean;
+    showInfoIcon?: boolean;
+    infoIconClassName?: string;
+    size?: TMediumSizes;
+    label: ReactNode | string;
+    labelClassName?: string;
+    onChange?: (
+        e:
+            | React.ChangeEvent<HTMLInputElement>
+            | React.KeyboardEvent<HTMLSpanElement>,
+    ) => void;
+    className?: string;
+}
+
+export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
+    (
+        {
+            checked = false,
+            disabled = false,
+            indeterminate = false,
+            showInfoIcon = false,
+            infoIconClassName,
+            size = "sm",
+            label,
+            labelClassName,
+            name,
+            onChange,
+            className,
+            ...rest
+        },
+        ref,
+    ) => {
+        const [isChecked, setIsChecked] = useState(checked);
+        const [isIndeterminate, setIsIndeterminate] = useState(indeterminate);
+
+        useEffect(() => {
+            setIsChecked(checked);
+        }, [checked]);
+
+        useEffect(() => {
+            setIsIndeterminate(indeterminate);
+        }, [indeterminate]);
```

Comment:
These useEffects are being triggered based on the state it is changing. It might create infinite loop.

Reply:
These `useEffects` are not being triggered based on the state they are changing. They are triggered if the props, which component receive from parent were changed. In my component I'm using `isChecked` from `const [isChecked, setIsChecked] = useState(checked);`